docs/database/sql.md

```diff
@@ -6,9 +6,34 @@ linkTitle: SQL
 
 **Status**: [Experimental][DocumentStatus]
 
-The SQL databases Semantic Conventions extend and override the [Database Semantic Conventions](database-spans.md)
-that describe common database operations attributes in addition to the Semantic Conventions
-described on this page.
+The SQL databases Semantic Conventions describes how common [Database Semantic Conventions](database-spans.md) apply to SQL databases.
+
+The following database systems (defined in the [`db.system`](./database-spans.md#notes-and-well-known-identifiers-for-dbsystem) set) are known to use SQL as their primary query language:
+
+- `cockroachdb`
+- `db2`
+- `derby`
+- `edb`
+- `ingres`
+- `interbase`
+- `firebird`
+- `h2`
+- `hsqldb`
+- `mariadb`
+- `maxdb`
+- `mssql`
+- `mssqlcompact`
+- `mysql`
+- `oracle`
+- `pervasive`
+- `postgresql`
+- `sqlite`
+- `trino`
+- `other_sql`
+
+Many other database systems provide SQL support and can be accessed via generic database driver APIs such as JDBC or ODBC.
+When system-specific conventions are defined for the database, and it's accessed using SQL, the instrumentation SHOULD
+follow database-specific conventions and MAY also adhere to SQL semantic conventions.
```

Comment:
this seems a little confusing, e.g. for `mssql` we are copying down all of the applicable sql semantic conventions to the system-specific convention

and for some other system, say cosmosdb, if it's accessed using jdbc, I think I would expect it to follow SQL semantic conventions (and not necessarily cosmosdb conventions)

Reply:
> mssql we are copying down all of the applicable sql semantic conventions to the system-specific convention

let me think how to make it more clear.

> and for some other system, say cosmosdb, if it's accessed using jdbc, I think I would expect it to follow SQL semantic conventions (and not necessarily cosmosdb conventions)

It's interesting.

E.g. with Cosmos users are still using CosmosDB SDK to query with SQL [e.g. this](https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/quickstart-java?pivots=devcontainer-codespace#query-items).

[Mongodb-jdbc driver](https://mvnrepository.com/artifact/org.mongodb/mongodb-jdbc/2.1.2) or [cassandra unofficial  jdbc driver](https://github.com/zhicwu/cassandra-jdbc-driver/tree/master) work on top of corresponding Mongo and Cassandra client libs.  If the underlying libraries are instrumented, the driver does not need to be, i.e. SQL conventions should not be expected.

If I understand correctly, your concern is for a different case when the driver is instrumented.

Do you think the following would work for both of our cases:

> Specific database client libraries SHOULD follow corresponding semantic conventions (if defined) and MAY also follow SQL conventions when database is being accessed via SQL.

> Instrumentations applied to generic SQL abstractions such as JDBC have limited knowledge about underlying databases and SHOULD adhere to SQL semantic conventions.

Reply:
> Instrumentations applied to specific database client libraries SHOULD follow corresponding database semantic conventions (if defined) and MAY also follow SQL conventions when database is being accessed via SQL.

can you give an example what this means in practice? e.g. how would a particular comsosdb span change if it was also following SQL conventions in [e.g. this case](https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/quickstart-java?pivots=devcontainer-codespace#query-items).[e.g. this](https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/quickstart-java?pivots=devcontainer-codespace#query-items)

> Instrumentations applied to generic SQL abstractions such as JDBC have limited knowledge about underlying databases and SHOULD adhere to SQL semantic conventions.

I like this clarification